dali/operators/image/crop/bbox_crop.cc

```diff
@@ -780,8 +801,18 @@ class RandomBBoxCropImpl : public OpImplBase<CPUBackend> {
 
         crop.crop = out_crop;
         crop.boxes.assign(bounding_boxes.begin(), bounding_boxes.end());
-        crop.bbox_indices.clear();  // indices will be populated by FilterByCentroid
-        FilterByCentroid(rel_crop, crop.boxes, crop.bbox_indices);
+        crop.bbox_indices.clear();  // indices will be populated by FilterBboxes
+        if (box_prune_method_ == BoxPruneMethod::Centroid) {
+          FilterBboxes(crop.boxes, crop.bbox_indices,
+            [&](const Box<ndim, float>& bbox_) { return rel_crop.contains(bbox_.centroid()); });
+        } else { // box_prune_method_ == BoxPruneMethod::RelativeThresh
+          FilterBboxes(crop.boxes, crop.bbox_indices,
+            [&](const Box<ndim, float>& bbox_) {
+              const float intersec_ = volume(intersection(rel_crop, bbox_));
+              return intersec_ / volume(bbox_) > bbox_prune_threshold_;
```

Comment:
```suggestion
          FilterBboxes(crop.boxes, crop.bbox_indices,
            [&](const Box<ndim, float>& bbox) { return rel_crop.contains(bbox.centroid()); });
        } else { // box_prune_method_ == BoxPruneMethod::RelativeThresh
          FilterBboxes(crop.boxes, crop.bbox_indices,
            [&](const Box<ndim, float>& bbox) {
              const float intersec = volume(intersection(rel_crop, bbox));
              return intersec / volume(bbox) > bbox_prune_threshold;
```
I don't think you need to append `_` as these are local variables

Reply:
The convention is to add trailing `_` for a class members variables, so how about just renaming it to something else to avoid the confusion?